lib/src/main/java/com/basistheory/elements/service/BasisTheoryElements.kt

```diff
@@ -90,6 +96,68 @@ class BasisTheoryElements internal constructor(
             throw ApiException(e.code, e.responseHeaders, e.responseBody, e.message)
         }
 
+    fun encryptTokens(
+        encryptTokenRequest: EncryptTokenRequest,
+    ): Array<EncryptTokenResponse> =
+        try {
+            val processedTokenRequests = processTokenRequests(encryptTokenRequest.tokenRequests)
+
+            processedTokenRequests.map { tokenData ->
+                val encrypted = JWEEncryption.encrypt(
+                    tokenData.first,
+                    encryptTokenRequest.publicKey,
+                    encryptTokenRequest.keyId
+                )
+                EncryptTokenResponse(encrypted, tokenData.second)
+            }.toTypedArray()
+        } catch (e: IllegalArgumentException) {
+            throw e
+        } catch (e: Exception) {
+            throw EncryptTokenException("Failed to encrypt tokens", e)
+        }
+
+    private fun processTokenRequests(tokenRequests: Any): List<Pair<Any, String>> {
+        val tokenRequestsMap = tokenRequests.toMap()
+
+        val isSingleTokenRequest = tokenRequestsMap.containsKey("type")
+        if (isSingleTokenRequest) {
+            return listOf(processIndividualTokenRequest(tokenRequestsMap))
+        }
+
+        return processMultipleTokenRequests(tokenRequestsMap)
+    }
```

Comment:
We could also make this more idiomatic/`kotlin-esque` if toMap is avaliable for tokenRequests

```kotlin
private fun processTokenRequests(tokenRequests: Any): List<Pair<Any, String>> =
    tokenRequests.toMap().let {
        listOfNotNull(
            processIndividualTokenRequest(it).takeIf { "type" in it }
        ) ?: processMultipleTokenRequests(it)
    }
```